### PR TITLE
Score writing in Clickhouse

### PR DESCRIPTION
**Pull request by maamalama**
Scores storing in Clickhouse

### CODE REVIEW

File: valhalla/jawn/src/controllers/public/requestController.ts [hunk 1]
```diff
@@ -165,7 +165,10 @@ export class RequestController extends Controller {
     @Request() request: JawnAuthenticatedRequest,
     @Path() requestId: string
   ): Promise<Result<null, string>> {
-    const scoreManager = new ScoreManager(request.authParams);
+    const scoreManager = new ScoreManager(
+      request.authParams.organizationId,
+      request.authParams
+    );
```
Comment (chitalian) [inline valhalla/jawn/src/controllers/public/requestController.ts:171]: why? if you are passing in authParams you already have org id there

File: valhalla/jawn/src/managers/score/ScoreManager.ts [hunk 2]
```diff
@@ -1,12 +1,68 @@
...
-import { err, Result } from "../../lib/shared/result";
+import { err, ok, Result, resultMap } from "../../lib/shared/result";
 import { BaseManager } from "../BaseManager";
+import { AuthParams } from "../../lib/db/supabase";
+import { a } from "js-tiktoken/dist/core-c3ffd518";
 
 export class ScoreManager extends BaseManager {
+  constructor(private orgId: string, authParams: AuthParams) {
+    super(authParams);
+  }
```
Comment (chitalian) [inline valhalla/jawn/src/managers/score/ScoreManager.ts:15]: delete plz

File: valhalla/jawn/src/managers/score/ScoreManager.ts [hunk 3]
```diff
@@ -57,4 +113,96 @@ export class ScoreManager extends BaseManager {
...
+      await clickhouseDb.dbQuery<InsertRequestResponseVersioned>(
+        `
+      SELECT *
+      FROM request_response_versioned
+      WHERE request_id = {val_0: UUID}
+      AND version = {val_1: UInt64}
+      AND organization_id = {val_2: String}
+      AND provider = {val_3: String}
+    `,
+        [newVersion.id, newVersion.version - 1, this.orgId, newVersion.provider]
```
Comment (chitalian) [inline valhalla/jawn/src/managers/score/ScoreManager.ts:133]: this.authparams.orgId should be ok here

**Review by chitalian — APPROVED** [review]

File: valhalla/jawn/src/managers/score/ScoreManager.ts (hunk 3 above)
Comment (maamalama) [reply valhalla/jawn/src/managers/score/ScoreManager.ts:133]: make sense!

File: valhalla/jawn/src/managers/score/ScoreManager.ts [hunk 4]
```diff
@@ -1,12 +1,65 @@
...
+        scores
+      );
+
+      if (supabaseRequest.error || !supabaseRequest.data) {
+        return err(supabaseRequest.error);
+      }
+
+      const request = await this.bumpRequestVersion(requestId);
+
+      if (request.error || !request.data) {
```
Comment (colegottdank) [inline valhalla/jawn/src/managers/score/ScoreManager.ts:40]: check if request.data.length > 0

File: valhalla/jawn/src/managers/score/ScoreManager.ts [hunk 5]
```diff
@@ -57,4 +110,101 @@ export class ScoreManager extends BaseManager {
       return err(error.message);
     }
   }
+
+  private async putScoresIntoClickhouse(newVersion: {
```
Comment (colegottdank) [inline valhalla/jawn/src/managers/score/ScoreManager.ts:114]: Copy paste db methods into ScoreStore (Also the bump method below)

**Review by colegottdank — APPROVED** [review]

**Pull request merged**